src/instruments/src/PFD/index.tsx

```diff
@@ -8,247 +10,218 @@ import { HeadingOfftape, HeadingTape } from './HeadingIndicator';
 import { AltitudeIndicatorOfftape, AltitudeIndicator } from './AltitudeIndicator';
 import { AirspeedIndicatorOfftape, AirspeedIndicator, MachNumber } from './SpeedIndicator';
 import { FMA } from './FMA';
-import { getSimVar, setSimVar, renderTarget, createDeltaTimeCalculator } from '../util.js';
+import { getSimVar, setSimVar } from '../util.js';
 import { SmoothSin, LagFilter, RateLimiter } from './PFDUtils';
 import { DisplayUnit } from '../Common/displayUnit';
 import { render } from '../Common';
 import './style.scss';
 
-/* eslint-disable max-len */
-// eslint-disable-next-line react/prefer-stateless-function
-class PFD extends Component {
-    private displayIndex: number;
-
-    private deltaTime: number;
-
-    private GetDeltaTime: () => number;
-
-    private prevAirspeed: number;
-
-    private VLs: number;
-
-    private barTimer: number;
-
-    private smoothFactor: number;
-
-    private isAttExcessive: boolean;
-
-    private AirspeedAccFilter: LagFilter;
-
-    private AirspeedAccRateLimiter: RateLimiter;
-
-    private LSButtonPressed: boolean;
+export const PFD: React.FC = () => {
+    const [displayIndex] = useState(() => {
+        const url = document.getElementsByTagName('a32nx-pfd')[0].getAttribute('url');
+        return url ? parseInt(url.substring(url.length - 1), 10) : 0;
+    });
```

Comment:
In my opinion this could shouldn't even live in the PFD, but one level above and supplied via a prop, but I'm not sure our React setup has any components above the PFD.

This diplayIndex concern seems like a concern greater than the PFD alone and could also apply to the ND or other displays.

Reply:
I agree. The `displayIndex` functionality is a recurring pattern in various displays. It's probably a good idea to standardise these patterns once all displays are migrated to React.